### DISCUSSION TITLE
The entry point to the CreateDXGIFactory2 procedure was not found in the DLL dxgi.dll

### DISCUSSION THREAD

**Discussion by mickhaorex** [tnodir/fort#36] [Q&A] [answered]
Windows 7 x64 system. After installation, at startup it asks for a file api-ms-win-shcore-scaling-l1-1-1.dll.
I downloaded two versions from https://www.dll.ru/files/api-ms-win-shcore-scaling-l1-1-1-dll and threw in %windir%\system32 and %windir%\syswow64, respectively. After that, at startup writes:

FortFirewall.exe - Entry point not found
The entry point to the CreateDXGIFactory2 procedure was not found in the DLL dxgi.dll .

How to make it work?

P.S. Greetings from Tashkent))

**Comment by tnodir** [comment, marked as answer]
On Windows 7 just install:
- [Visual C++ x86 (32-bit!) redistributable](https://aka.ms/vs/17/release/vc_redist.x86.exe)
- [FortFirewall-3.5.8-windows-x86.exe](https://github.com/tnodir/fort/releases/download/v3.5.8/FortFirewall-3.5.8-windows-x86.exe)

**Reply by mickhaorex** [reply]
Thanks for the detailed answer!
Is it possible to embed all the necessary dll files in an exe file? To make your program work without the need to install additional libraries?

**Reply by tnodir** [reply]
> Is it possible to embed all the necessary dll files in an exe file?

No, thanks. See [Requirements](https://github.com/tnodir/fort#requirements).
